fs/localfs/local_fs_nonwindows.go

```diff
@@ -18,3 +18,14 @@ func platformSpecificOwnerInfo(fi os.FileInfo) fs.OwnerInfo {
 
 	return oi
 }
+
+func platformSpecificDeviceInfo(fi os.FileInfo) fs.DeviceInfo {
+	var oi fs.DeviceInfo
+	if stat, ok := fi.Sys().(*syscall.Stat_t); ok {
```

Comment:
@eras AFAICT, this is NOT incurring a separate `stat` syscall, and instead simply reusing and casting the result of a previous call, so we may be OK here.